ecosystem/typescript/sdk/src/tests/e2e/client.test.ts

```diff
@@ -1,178 +1,235 @@
-import { NODE_URL } from "../unit/test_helper.test";
+import { getTransaction, longTestTimeout, NODE_URL } from "../unit/test_helper.test";
 
-test("server response should include cookies", async () => {
-  try {
-    const response = await aptosRequest({
-      // use devnet as localnet doesnt set cookies
-      url: "https://fullnode.devnet.aptoslabs.com/v1",
-      method: "GET",
-      originMethod: "test cookies",
-    });
-    expect(response.headers).toHaveProperty("set-cookie");
-  } catch (error: any) {
-    // should not get here
-    expect(true).toBe(false);
-  }
-});
+test(
+  "server response should include cookies",
+  async () => {
+    try {
+      const response = await aptosRequest({
+        // use devnet as localnet doesnt set cookies
+        url: "https://fullnode.devnet.aptoslabs.com/v1",
+        method: "GET",
+        originMethod: "test cookies",
+      });
+      expect(response.headers).toHaveProperty("set-cookie");
+    } catch (error: any) {
+      // should not get here
+      expect(true).toBe(false);
+    }
+  },
+  longTestTimeout,
+);
 
-test("call should include x-aptos-client header", async () => {
-  try {
-    const response = await aptosRequest({
-      url: `${NODE_URL}`,
-      method: "GET",
-      endpoint: "accounts/0x1",
-      body: null,
-      originMethod: "test x-aptos-client header",
-    });
-    expect(response.config.headers).toHaveProperty("x-aptos-client", `aptos-ts-sdk/${VERSION}`);
-  } catch (error: any) {
-    // should not get here
-    expect(true).toBe(false);
-  }
-});
+test(
+  "call should include x-aptos-client header",
+  async () => {
+    try {
+      const response = await aptosRequest({
+        url: `${NODE_URL}`,
+        method: "GET",
+        endpoint: "accounts/0x1",
+        body: null,
+        originMethod: "test x-aptos-client header",
+      });
+      expect(response.config.headers).toHaveProperty("x-aptos-client", `aptos-ts-sdk/${VERSION}`);
+    } catch (error: any) {
+      // should not get here
+      expect(true).toBe(false);
+    }
+  },
+  longTestTimeout,
+);
 
-test("when token is set", async () => {
-  try {
-    const response = await aptosRequest({
-      url: `${NODE_URL}`,
-      method: "GET",
-      endpoint: "accounts/0x1",
-      body: null,
-      originMethod: "test 200 status",
-      overrides: { TOKEN: "my-token" },
-    });
-    expect(response.config.headers).toHaveProperty("Authorization", "Bearer my-token");
-  } catch (error: any) {
-    // should not get here
-    expect(true).toBe(false);
-  }
-});
+test(
+  "call should include all expected headers",
+  async () => {
+    const bcsTxn = await getTransaction();
+    try {
+      const response = await aptosRequest({
+        url: `${NODE_URL}`,
+        method: "POST",
+        endpoint: "transactions",
+        body: bcsTxn,
+        originMethod: "test request includes all headers",
+        mediaType: "application/x.aptos.signed_transaction+bcs",
```

Comment:
I see, so when the request is a bcs serialized txn, the content type need to be set to `application/x.aptos.signed_transaction+bcs`? Hmm not sure if I seen this in Python nor Rust sdk

Reply:
interesting.... we should have it, without it API explodes lol

Reply:
found it https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/python/sdk/aptos_sdk/async_client.py#L245

Reply:
ohhh nice find!